### PR TITLE
CSHARP-4202: Specify correct max supported wire version.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**

### CODE REVIEW

File: tests/MongoDB.Driver.Core.TestHelpers/XunitExtensions/RequireServer.cs [hunk 1]
```diff
@@ -140,6 +140,19 @@ public RequireServer Serverless(bool require = true)
...
+        {
+            var serverVersion = CoreTestConfiguration.ServerVersion;
+            var isStableServer = serverVersion.PreRelease == null;
+
+            if (isStableServer == stable)
+            {
+                return this;
+            }
+
+            throw new SkipException("Test skipped because expected server should be in " + (stable ? "GA" : "prerelease") + $" state, but found {serverVersion}.");
```
Comment (BorisDog) [inline tests/MongoDB.Driver.Core.TestHelpers/XunitExtensions/RequireServer.cs:153]: minor: maybe "because {GA/prerelease} server version is expected"?

**Review by BorisDog — COMMENTED** [review]
Some minor comments

File: tests/MongoDB.Driver.Core.TestHelpers/XunitExtensions/RequireServer.cs (hunk 1 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Core.TestHelpers/XunitExtensions/RequireServer.cs:153]: done

File: tests/MongoDB.Driver.Core.Tests/Core/Servers/ServerDescriptionTests.cs [hunk 2]
```diff
@@ -277,7 +277,7 @@ public void Equals_should_return_true_when_all_fields_are_equal()
         [InlineData(new[] { 13, 15 }, true)]
         [InlineData(new[] { 14, 15 }, true)]
         [InlineData(new[] { 15, 16 }, true)]
-        [InlineData(new[] { 16, 17 }, false)]
+        [InlineData(new[] { 18, 19 }, false)]
```
[the remark this answers is not in the document]
Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Core.Tests/Core/Servers/ServerDescriptionTests.cs:280]: done

**Review by BorisDog — APPROVED** [review]
LGTM

**Pull request merged**